imf/distributions.py

```diff
@@ -109,23 +109,28 @@ def __init__(self, slopes, breaks):
 
     def pdf(self, x):
         x1 = np.asarray(x)
-        ret = x1 * 0.
+        ret = np.atleast_1d(x1) * 0.
         for ii in range(self.nsegm):
-            xind = (x1 < self.breaks[ii + 1]) & (x1 > self.breaks[ii])
+            xind = (x1 < self.breaks[ii + 1]) & (x1 >= self.breaks[ii])
             if xind.sum() > 0:
                 ret[xind] = self.weights[ii] * self.pows[ii].pdf(x1[xind])
-        return ret
+        return ret.reshape(x1.shape)
 
     def cdf(self, x):
         x1 = np.asarray(x)
-        ret = x1 * 0.
+        ret = np.atleast_1d(x1) * 0.
         cums = np.r_[[0], np.cumsum(self.weights)]
         for ii in range(self.nsegm):
-            xind = (x1 < self.breaks[ii + 1]) & (x1 > self.breaks[ii])
+            xind = (x1 < self.breaks[ii + 1]) & (x1 >= self.breaks[ii])
             if xind.sum() > 0:
                 ret[xind] = cums[ii] + self.weights[ii] * self.pows[ii].cdf(
                     x1[xind])
-        return ret
+        xind = x1>self.breaks[-1]
+        if xind.sum():
```

Comment:
this is testing `if xind.sum() != 0`, right?  It might be better to make that explicit; I'm not sure, but I think numpy will sometimes warn if you try to treat a float as a boolean

Reply:
Yes, it was a typo. I fixed that